From f2b0ddde483e317b8e101c93d17ab96b5463ab94 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Julian=20R=C3=BCth?= <julian.rueth@fsfe.org>
Date: Wed, 9 Feb 2022 18:18:40 -0500
Subject: [PATCH] Fix import order in setup.py

Importing Setuptools breaks some of distutils so they need to be
imported in the correct order.
---
 pyeantic/src/setup.py.in | 19 +++++++++++--------
 1 file changed, 11 insertions(+), 8 deletions(-)

diff --git a/pyeantic/src/setup.py.in b/pyeantic/src/setup.py.in
index 1e9ff49..d2c951f 100644
--- a/pyeantic/src/setup.py.in
+++ b/pyeantic/src/setup.py.in
@@ -1,12 +1,14 @@
 import os
-from distutils.core import setup
 from setuptools.command.egg_info import egg_info
+from distutils.core import setup
+
 
 class vpath_egg_info(egg_info):
     r"""
     A VPATH aware egg_info for VPATH builds with Automake, see
     https://blog.kevin-brown.com/programming/2014/09/24/combining-autotools-and-setuptools.html
     """
+
     def run(self):
         r"""
         Run egg_info.run() but build the .egg-info in the --build-base path
@@ -18,16 +20,17 @@ class vpath_egg_info(egg_info):
             self.egg_info = os.path.join(self.egg_base, os.path.basename(self.egg_info))
         egg_info.run(self)
 
+
 setup(
-    name = 'pyeantic',
-    version = '@PACKAGE_VERSION@',
-    packages = ['pyeantic',],
-    license = 'GPL 2.0+',
+    name='pyeantic',
+    version='@PACKAGE_VERSION@',
+    packages=['pyeantic'],
+    license='GPL 2.0+',
     install_requires=[
         'cppyy'
     ],
-    long_description = open('@abs_top_srcdir@/../README.md').read(),
+    long_description=open('@abs_top_srcdir@/../README.md').read(),
     include_package_data=True,
-    cmdclass={ 'egg_info': vpath_egg_info },
-    package_dir={ "": os.path.relpath('@abs_top_srcdir@/src/') },
+    cmdclass={'egg_info': vpath_egg_info},
+    package_dir={"": os.path.relpath('@abs_top_srcdir@/src/')},
 )
-- 
2.35.1

